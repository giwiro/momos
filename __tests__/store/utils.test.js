// @flow
import {
  ActionCreator,
  generateActionCreators,
  createReducer,
} from '../../src/store/utils';

describe('ActionCreator', () => {
  it('should return correct class name as type property', () => {
    const ac = new ActionCreator();
    expect(ac.type).toMatchSnapshot();
  });

  it('should return correct function name', () =>
    expect(ActionCreator.fnName).toMatchSnapshot());

  it('should return an appended payload', () =>
    expect(new ActionCreator({foo: 'bar'})).toMatchSnapshot());
});

describe('generateActionCreators', () => {
  it('should generate function map from generate action creators', () => {
    const gen = generateActionCreators([ActionCreator]);
    expect(gen).toMatchSnapshot();
  });

  it('should execute the generated function', () => {
    const gen = generateActionCreators([ActionCreator]);
    expect(gen.actionCreator()).toMatchSnapshot();
  });
});

describe('createReducer', () => {
  const reducerMap = {
    ActionCreator: () => ({foo: 'bar'}),
  };

  it('should create the reducer', () =>
    expect(createReducer({}, reducerMap)).toMatchSnapshot());

  it('should execute correctly the reducer', () =>
    expect(
      createReducer({}, reducerMap)({}, new ActionCreator()),
    ).toMatchSnapshot());
});
