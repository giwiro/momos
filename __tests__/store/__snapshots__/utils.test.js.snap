// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCreator should return an appended payload 1`] = `
Object {
  "foo": "bar",
  "type": "ActionCreator",
}
`;

exports[`ActionCreator should return correct class name as type property 1`] = `"ActionCreator"`;

exports[`ActionCreator should return correct function name 1`] = `"actionCreator"`;

exports[`createReducer should create the reducer 1`] = `[Function]`;

exports[`createReducer should execute correctly the reducer 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`generateActionCreators should execute the generated function 1`] = `
Object {
  "type": "ActionCreator",
}
`;

exports[`generateActionCreators should generate function map from generate action creators 1`] = `
Object {
  "actionCreator": [Function],
}
`;
